### PR TITLE
Implementation of MMU-functionalities

### PR DESCRIPTION
**Pull request by js97nMH**
### RET - Instructions:
Implementation of a privilege-check that will identify a context-switch.
When a context-switch is detected, a return variable `ret` is set with the value -2, which corresponds to `etiss::RETURNCODE::RELOADBLOCKS`.
The return statement `return exception;` that is generated by the code generator needs to be replaced by: `return ret;`
### CSR - Instructions:
Added a blackbox function which notifies the MMU of a change to the SATP-register: `extern signed<32> ETISS_SIGNAL_MMU(unsigned<XLEN> mmu_signal_) [[etiss_needs_arch]];`
Change to csr_write() funtion: if the csr specified by the instruction is the SATP-csr, ETISS_SIGNAL_MMU() is called.
### SFENCE_VMA - Instruction:
Implementation of SFENCE.VMA. The instruction behavior defines 4 cases as mentioned by the RISCV-specification. Each case will call a respective function to communicate the memory fence to the MMU.
The function `evict_all()` calls a blackbox function that flushes the TLB in the MMU.
The functions `evict_asid(asid)` and `evict_addr_asid(vaddr, asid)` also cause a full TLB-flush. New blackbox-functions that permit targetet eviction should be implemented at a later stage.
The function `evict_addr(vaddr)` calls a blackbox function that only evicts the specified vaddr from the TLB.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "rv_base"]
 	path = rv_base
-	url = ../RISCV_ISA_CoreDSL
```
Comment (wysiwyng) [inline .gitmodules:3]: Is this necessary? `../repo` also works wih URLs, see here: https://git-scm.com/docs/git-submodule

File: tum_mod.core_desc [hunk 2]
```diff
@@ -7,7 +7,13 @@ InstructionSet tum_ret extends RISCVBase {
         URET [[no_cont]] {
             encoding: 0b0000000 :: 0b00010 :: 0b00000 :: 0b000 :: 0b00000 :: 0b1110011;
             behavior: {
-				CSR[3088] = 0;
+                // priviledge change indicates context-switch
```
Comment (wysiwyng) [inline tum_mod.core_desc]: typo: privilege

File: tum_mod.core_desc [hunk 3]
```diff
@@ -60,6 +82,10 @@ InstructionSet tum_csr extends Zicsr {
             else if (csr == FRM_N) CSR[FCSR_N] = ((val & 0x07) << 5) | (CSR[FCSR_N] & 0x1F);
             else if (csr == FCSR_N) CSR[FCSR_N] = val & 0xFF;
             else CSR[csr] = val;
+            // evaluate if changes have been made to SATP
+            if (csr == SATP) {
+                signed<32> tmp = ETISS_SIGNAL_MMU(val);
+            }
```
Comment (wysiwyng) [inline tum_mod.core_desc]: integrate this into the `if .. else if` structure

**Review by wysiwyng — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (js97nMH) [reply .gitmodules:3]: I don't think this is necessary, definitively not for the main repository. But I had trouble fetching the submodule in my fork, so I had to change this for myself.

Comment (wysiwyng) [reply .gitmodules:3]: that makes sense of course, leave this in for the moment.

File: tum_mod.core_desc (hunk 2 above)
Comment (js97nMH) [reply tum_mod.core_desc]: thanks, will change it

Comment (js97nMH) [reply tum_mod.core_desc]: Fixed the typos

File: tum_mod.core_desc (hunk 3 above)
Comment (js97nMH) [reply tum_mod.core_desc]: I integrated it into the `if ... else if` structure.

**Comment by wysiwyng** [comment]
merged into https://github.com/tum-ei-eda/etiss_arch_riscv/tree/coredsl_mmu

**Pull request closed without merging**